tests/e2e/framework/utils.go

```diff
@@ -384,6 +386,134 @@ func (f *Framework) AssertMetricsEndpointUsesHTTPVersion(endpoint, version strin
 	return nil
 }
 
+func runOCandGetOutput(arg []string) (string, error) {
+	ocPath, err := exec.LookPath("oc")
+	if err != nil {
+		return "", fmt.Errorf("Failed to find oc binary: %v", err)
+	}
+
+	cmd := exec.Command(ocPath, arg...)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("Failed to run oc command: %v", err)
+	}
+	return string(out), nil
+}
+
+// createServiceAccount creates a service account
+func (f *Framework) SetupRBACForMetricsTest() error {
+	_, err := runOCandGetOutput([]string{
+		"create", "sa", PromethusTestSA, "-n", f.OperatorNamespace})
+	if err != nil {
+		return fmt.Errorf("Failed to create service account: %v", err)
+	}
+
+	_, err = runOCandGetOutput([]string{
+		"adm", "policy", "add-cluster-role-to-user", "cluster-monitoring-view", "-z", PromethusTestSA, "-n", f.OperatorNamespace})
+	if err != nil {
+		return fmt.Errorf("Failed to add cluster role to user: %v", err)
+	}
+	return nil
+}
+
+// CleanupRBACForMetricsTest deletes the service account
+func (f *Framework) CleanUpRBACForMetricsTest() error {
+	_, err := runOCandGetOutput([]string{
+		"delete", "sa", PromethusTestSA, "-n", f.OperatorNamespace})
+	if err != nil {
+		return fmt.Errorf("Failed to delete service account: %v", err)
+	}
+	return nil
+}
+
+// GetPrometheusMetricTargets retrieves Prometheus metric targets
+func (f *Framework) GetPrometheusMetricTargets() ([]promv1.Target, error) {
+	var metricsTargets []promv1.Target
+	const prometheusCommand = `TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token) && { curl -k -s https://prometheus-k8s.openshift-monitoring.svc.cluster.local:9091/api/v1/targets --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $TOKEN"; }`
+	namespace := f.OperatorNamespace
+	out, err := runOCandGetOutput([]string{
+		"run", "--rm", "-i", "--restart=Never", "--image=registry.fedoraproject.org/fedora:latest",
```

Comment:
We could probably get away with only fedora-minimal here, but we can patch that after.